zhmcclient/_partition.py

```diff
@@ -1550,6 +1550,529 @@ def list_attached_storage_groups(self, full_properties=False):
+
+          number_of_nics (int): Number of NICs to be created for the partition
+            as endpoints for the partition link.
+
+          nic_property_list (list of dict): List whose items specify the
+            properties for the NICs to be created. `None` is treated as an empty
+            list. The list must have no more items than the number of NICs to be
+            created. NICs beyond the end of the list are created with default
+            properties.
+
+            The following NIC properties may be specified in the list items:
+
+            * ``device-number`` (string): Device number. For Hipersockets, 3
+              consecutive device numbers are used, starting with the specified
+              device number. For SMC-D, the specified device number is used
+              as the single device number. Default: auto-generated.
+            * ``fid`` (int): SMC-D only: FID. Default: auto-generated.
+            * ``vlan-id`` (int): Hipersockets only: VLAN ID. None means no
+              VLAN is used. Default: No VLAN is used.
+            * ``mac-address`` (string): Hipersockets only: MAC address.
+              Default: auto-generated.
+
+          wait_for_completion (bool):
+            Boolean controlling whether this method should wait for completion
+            of the asynchronous job on the HMC.
+
+          operation_timeout (:term:`number`):
+            Timeout in seconds, for waiting for completion of the asynchronous
+            job on the HMC. The special value 0 means that no timeout is set.
+            `None` means that the default async operation timeout of the
+            session is used. If the timeout expires when
+            `wait_for_completion=True`, a :exc:`~zhmcclient.OperationTimeout`
+            is raised.
+
+        Returns:
+
+          * If `wait_for_completion` is `True`, returns `None`.
+
+          * If `wait_for_completion` is `False`, returns a
+            :class:`~zhmcclient.Job` object representing the asynchronously
+            executing job on the HMC.
+            This job does not support cancellation.
+
+        Raises:
+
+          :exc:`~zhmcclient.PartitionLinkError`
+          :exc:`~zhmcclient.HTTPError`
+          :exc:`~zhmcclient.ParseError`
+          :exc:`~zhmcclient.AuthError`
+          :exc:`~zhmcclient.ConnectionError`
+          :exc:`~zhmcclient.OperationTimeout`: The timeout expired while
+            waiting for completion of the operation.
+
+        Examples:
+
+          * Attach a Hipersockets-type partition link to a partition, and
+            specify device number and VLAN::
+
+                part1.attach_network_link(
+                    hs_link, nic_property_list=[
+                        {
+                            'device-number': '1300',
+                            'vlan-id': '53',
+                        }
+                    ])
+        """
+        body = {}
+        # pl_type = partition_link.get_property('type')
```

Comment:
Why is this code commented out?

Reply:
I think this was a remainder from the initial implementation of the attach method where it tried to support both CTC and network links. I have removed it. DONE.